### DISCUSSION TITLE
Marginal effects for sample selection model

### DISCUSSION THREAD

**Discussion by reddylee** [vincentarelbundock/marginaleffects#1179] [Q&A]
Hello,
I'm encountering challenges in calculating marginal effects for a sample selection model using the sampleSelection package. Currently, I can only obtain marginal effects for the selection equation. I'm wondering if it's possible to calculate marginal effects for the outcome equation as well. Many thanks.

**Comment by vincentarelbundock** [comment]
Have you tried the `type` argument? If so, please supply a minimal replicable example and explain exactly why the `type` argument fails. 

Here are some tips on asking good questions on software help forums: https://marginaleffects.com/vignettes/help.html#getting-help

**Reply by reddylee** [reply]
Thanks for the quick reply. I tried type argument:

type = "link" get the coefficients for selection equation
type = "unconditional" get the coefficients for outcome equation
While type = "response" (which is the default) only get one outcome of marginal effects, which I have no idea it is for selection or for outcome.

```
library(sampleSelection)
library(marginaleffects)
data(Mroz87)

m <- selection(lfp ~ educ + age + kids5 + kids618 + nwifeinc,
               wage >= 5 ~ educ + exper, data = Mroz87)

avg_slopes(m) |> 
  view()
```

**Reply by vincentarelbundock** [reply]
Thanks for the example.

I looked into this and the issue is that you need to specify *both* the `type` argument and the `part` argument. This is documented in `?predict.sampleSelection`

```r
avg_slopes(m, type = "unconditional", part = "outcome")
avg_slopes(m, type = "response", part = "selection")
avg_slopes(m, type = "link", part = "selection")
```

**Reply by reddylee** [reply]
Thanks, Vincent. Your code can not get the marginal effects of the outcome. Please see the comments on the code below. `?predict.sampleSelection` does not work, it reads "No documentation for ‘predict.sampleSelection’ in specified packages and libraries:". I did not find predict.sampleSelection in the codebook (https://cran.r-project.org/web/packages/marginaleffects/marginaleffects.pdf, June 14 2024). Do I miss something?


```
avg_slopes(m, type = "unconditional", part = "outcome")#coefficients of outcome

avg_slopes(m, type = "response", part = "selection")# marginal effects of selection

avg_slopes(m, type = "link", part = "selection")#coefficients of selection
```

**Reply by vincentarelbundock** [reply]
oops, it's `?sampleSelection::predict.selection` and in the `sampleSelection` package codebook.

**Reply by vincentarelbundock** [reply]
What do you mean by "coefficient of outcome"? The outcome component is linear, so the slopes correspond to the coefficients.

**Comment by reddylee** [comment]
Thanks, Vincent. Yes, for normal selection model, the outcome equation is linear. However, I am using two binary model, which both selection and outcome equation are probit model. Therefore, the outcome model is not linear. I guess that's the reason we get the same. Wonder if it can deal with this double probit model.

**Reply by vincentarelbundock** [reply]
This is not a `marginaleffects` question. If the functionality is supported through the `type` argument in the `sampleSelection` package, then it should be supported in `marginaleffects` as well.
